tests/test_es_doc_publisher.py

```diff
@@ -636,6 +636,46 @@ async def test_state_to_bulk_action_via_initialized_publisher(
             } == snapshot
 
 
+class Test_Benchmark_Tests:
+    """Benchmark tests for the Elasticsearch Document Publisher."""
+
+    @pytest.fixture(autouse=True)
+    def freeze_time(freezer: FrozenDateTimeFactory):
+        """Do not freeze time, override auto-use fixture."""
+        return
+
+    @pytest.mark.asyncio
+    async def test_publishing_benchmark(
+        self,
+        hass: HomeAssistant,
+        state,
+        initialized_publisher: DocumentPublisher,
+        es_aioclient_mock: AiohttpClientMocker,
+    ):
+        """Benchmark entity publishing."""
+
+        # If processing 500 entities takes more than 1s we have a significant performance degration
```

Comment:
How did you arrive at the 500 / 1s benchmark? I worry could miss significant regressions, as it still passes on my machine when I increase it by nearly 3 orders of magnitude (500 -> 100,000).

Reply:
I wanted to have something first and make sure it plays nice in CI, doesn't flap, etc but I'm happy to bump it.

My plan was to follow-up later probably using the official pytest benchmark library or something.

Maybe ill parametrize it and have it test like 50, 500, 5000, and 50,000 and snapshot the duration and just make sure it's within some% of snapshot?

Reply:
> I wanted to have something first and make sure it plays nice in CI, doesn't flap, etc but I'm happy to bump it.
> 
> My plan was to follow-up later probably using the official pytest benchmark library or something.

I'm good with this approach.

> Maybe ill parametrize it and have it test like 50, 500, 5000, and 50,000 and snapshot the duration and just make sure it's within some% of snapshot?

I don't think it's worth the complexity of rolling our own if you plan to use something like the pytest benchmark library in a follow-up